packages/components/src/Components/ConditionalFilter/Group.js

```diff
@@ -229,21 +234,53 @@ class Group extends Component {
 
     render() {
         const { isExpanded, filterBy } = this.state;
-        const { groups, items, placeholder, className, selected, isFilterable, isDisabled, onFilter, onShowMore, showMoreTitle, showMoreOptions } = this.props;
+        const {
+            groups,
+            items,
+            placeholder,
+            className,
+            selected,
+            isFilterable,
+            isDisabled,
+            maxFilterItems,
+            onFilter,
+            isFilterExpandable,
+            onShowChange,
+            showMoreTitle,
+            showLessTitle,
+            showMoreOptions } = this.props;
         const filterItems = items || groups;
 
         const showMore = {
             type: groupType.button,
             variant: showMoreOptions?.variant || 'link',
             items: [{
                 ...showMoreOptions,
-                label: showMoreTitle,
+                label: this.state.isCollapsed ? showMoreTitle : showLessTitle,
                 type: groupType.button,
-                onClick: onShowMore,
-                isPersistentAction: true
+                onClick: onShowChange ? onShowChange : this.toggleCollapsed
             }]
         };
 
+        const { groups: filteredGroups, itemTotal } = groups.reduce((acc, group) => {
```

Comment:
You are checking `isFilterExpandable` at several places in this reduce, I think you can extract it out.